frontend/public/components/alert.tsx

```diff
@@ -0,0 +1,427 @@
+                  <dt>Alert Rule</dt>
+                  <dd>
+                    <div className="co-resource-link">
+                      <ResourceIcon kind={ruleKind} />
+                      <Link to={detailsURL('AlertRule', name, _.get(rule, 'labels'))} className="co-resource-link__resource-name">{name}</Link>
+                    </div>
+                  </dd>
+                </dl>
+              </div>
+              <div className="col-sm-6">
+                <dl className="co-m-pane__details">
+                  <Annotation title="Description">{annotations.description}</Annotation>
+                  <Annotation title="Summary">{annotations.summary}</Annotation>
+                  <Annotation title="Message">{annotations.message}</Annotation>
+                  {annotations.runbook_url && <React.Fragment>
+                    <dt>Runbook</dt>
+                    <dd><ExternalLink href={annotations.runbook_url} text={annotations.runbook_url} /></dd>
+                  </React.Fragment>}
+                </dl>
+              </div>
+            </div>
+          </div>
+        </div>
+      </StatusBox>
+    </React.Fragment>;
+  }
+}
+export const AlertsDetailsPage = connectPage(connect(alertStateToProps)(AlertsDetailsPage_));
+
+const ViewInPrometheusLink_ = ({rule, urls}) => {
+  const baseUrl = urls[MonitoringRoutes.Prometheus];
+  const query = _.get(rule, 'query');
+  if (!baseUrl || !query) {
+    return null;
+  }
+  const href = `${baseUrl}/graph?g0.expr=${encodeURIComponent(query)}&g0.tab=0`;
+  return <ExternalLink href={href} text="View in Prometheus UI" />;
+};
+const ViewInPrometheusLink = connectToPrometheusURL(ViewInPrometheusLink_);
+
+const ActiveAlerts = ({alerts}) => <div className="co-m-table-grid co-m-table-grid--bordered">
+  <div className="row co-m-table-grid__head">
+    <div className="col-xs-6">Description</div>
+    <div className="col-xs-2">Active Since</div>
+    <div className="col-xs-2">State</div>
+    <div className="col-xs-2">Value</div>
+  </div>
+  <div className="co-m-table-grid__body">
+    {alerts.map((a, i) => {
+      const name = a.labels.alertname;
+      const description = _.get(a, 'annotations.description') || _.get(a, 'annotations.message') || name;
+      return <ResourceRow key={i} obj={a}>
+        <div className="col-xs-6 co-resource-link-wrapper">
+          <Link className="co-resource-link" to={detailsURL('Alert', name, a.labels)}>{description}</Link>
+        </div>
+        <div className="col-xs-2"><Timestamp timestamp={a.activeAt} /></div>
+        <div className="col-xs-2"><State state={a.state} /></div>
+        <div className="col-xs-2">{a.value}</div>
+      </ResourceRow>;
+    })}
+  </div>
+</div>;
+
+const ruleStateToProps = (ignore, {match, rules}) => {
+  const name = _.get(match, 'params.name');
+  const rule = _.find(rules, {name, labels: getURLSearchParams()});
+  return {name, rule};
+};
+
+class AlertRulesDetailsPage_ extends AlertsPageWrapper {
```

Comment:
Skeptical of the need to use inheritance, which is usually discouraged in React because it muddies the lifecycle hooks. Prefer composition instead.